File: Java/src/Shape/Point2D.java
package Shape;

public class Point2D {
    private double x;
    private double y;

    public Point2D() {
        x = 0;
        y = 0;
    }

    public Point2D(double _x) {
        x = _x;
        y = 0;
    }

    public Point2D(double _x, double _y) {
        this(_x);
        y = _y;
    }

    public Point2D(Point2D p2d) {
        x = p2d.x;
        y = p2d.y;
    }

    public double getX() {
        return x;
    }

    public double getY() {
        return y;
    }

    public void setX(double nx) {
        x = nx;
    }

    public void setY(double ny) {
        y = ny;
    }

    public static double distance(Point2D p1, Point2D p2) {
        return Math.sqrt(Math.pow(p2.x - p1.x, 2) + Math.pow(p2.y - p1.y, 2));
    }

    public boolean equals(Object obj) {
        if (this == obj)
            return true;
        if (obj != null && obj.getClass() == getClass()) {
            return x == ((Point2D) obj).x && y == ((Point2D) obj).y;
        }
        return false;
    }

    public String toString() {
        return "(" + x + "," + y + ")";
    }

}

File: cpp/KalkCpp/Shape/point2d.cpp
#include "point2d.h"
#include "math.h"
#include <typeinfo>
#include <string>


Point2D::Point2D(double var_x, double var_y) : x(var_x), y(var_y)
{

}



bool Point2D::operator==(const Point2D& p) const
{
    if(this == &p)
        return true;
    if(&p != 0 && typeid(*this) == typeid(p))
    {
        return x == p.x && y == p.y;
    }
    return false;
}




double Point2D::distance(const Point2D& p1, const Point2D& p2)
{
    return sqrt(pow(p2. x - p1. x, 2) + pow(p2.y - p1.y, 2));
}



double Point2D::getX() const
{
    return x;
}



double Point2D::getY() const
{
    return y;
}


void Point2D::setX(double nx)
{
    x = nx;
}

void Point2D::setY(double ny)
{
    y = ny;
}

void Point2D::printPoint() const
{
     std::cout << "P2D(" + std::to_string(x) + "," + std::to_string(y) + ")";
}

std::ostream& operator<<(std::ostream& os, const Point2D& p2d)
{
    return os << "(" << p2d.x << "," << p2d.y << ")";
}


Point2D::~Point2D()
{

}
